python/ray/tune/ray_trial_executor.py

```diff
@@ -87,10 +90,13 @@ def _stop_trial(self, trial, error=False, error_msg=None,
             stop_logger (bool): Whether to shut down the trial logger.
         """
 
+        if stop_logger:
```

Comment:
Why do this earlier?

Reply:
Oh, this is so that when the trial checkpoint state is updated at `set_status`, the `result_logger` is None already.